src/shogun/base/SGObject.h

```diff
@@ -382,10 +382,11 @@ class CSGObject
 		{
 			return any_cast<T>(value);
 		}
-		catch (const std::logic_error& exc)
+		catch (const TypeMismatchException& exc)
 		{
 			SG_ERROR(
-				"Get \"%s\" failed: %s.\n", _tag.name().c_str(), exc.what());
+				"Get \"%s\" failed. Expected %s, got %s.\n",
```

Comment:
I would maybe give a little more info:
"Getting parameter parameter CLASSNAME::PARAMNAME failed. Requested type is X, but internal type is Y"